### PR TITLE
Datasource trigger integration test

### PR DESCRIPTION
**Pull request by jenswaechtler**
The integration test for the datasource trigger endpoint. There is a new endpoint for the mock server that dynamically responds to a URI. The test is done once with free variables and once without free variables with an empty POST request.
@georg-schwarz @mathiaszinnen

### CODE REVIEW

**Comment by jenswaechtler** [comment]
On my local machine the test runs without errors. Do you know what the problem could be?

File: adapter/integration-test/src/adapter-datasource.test.js [hunk 1]
```diff
@@ -184,6 +187,52 @@ describe('Adapter Configuration', () => {
...
 })
 
+test('POST datasources/{id}/trigger', async () => {
+  const dynamicDatasourceResponse = await request(URL)
+    .post('/datasources')
+    .send(dynamicDatasourceConfig)
+  const dynDatasourceId = dynamicDatasourceResponse.body.id
+
+  const dynamicDataMetaData = await request(URL)
+    .post('/datasources/' + dynDatasourceId + '/trigger')
```
Comment (georg-schwarz) [inline adapter/integration-test/src/adapter-datasource.test.js]: Little pointer to how you can concat strings very nicely in JavaScript:
```
`/datasources/${dynDatasourceId}/trigger`
```

File: adapter/integration-test/src/adapter-datasource.test.js [hunk 2]
```diff
@@ -184,6 +187,52 @@ describe('Adapter Configuration', () => {
   })
 })
 
+test('POST datasources/{id}/trigger', async () => {
```
Comment (georg-schwarz) [inline adapter/integration-test/src/adapter-datasource.test.js]: Does it make sense to split dynamic and non-dynamic into separate test cases?

**Comment by georg-schwarz** [comment]
Integrationtest for adapter red, internal server error!

File: adapter/integration-test/src/adapter-datasource.test.js (hunk 2 above)
Comment (jenswaechtler) [reply adapter/integration-test/src/adapter-datasource.test.js]: Maybe, but actually it should be explicit for the trigger endpoint. I followed the other tests that test for the endpoint as well.

File: adapter/integration-test/src/adapter-datasource.test.js (hunk 1 above)
Comment (jenswaechtler) [reply adapter/integration-test/src/adapter-datasource.test.js]: Thanks for the suggestion. I changed it right away.

**Comment by mathiaszinnen** [comment]
> On my local machine the test runs without errors. Do you know what the problem could be?

I'll look into it later

**Review by mathiaszinnen — COMMENTED** [review]
The internal server error is probably caused by rabbitmq. I had the same problem before. RabbitMQ is started as a dependency but the trigger test runs before it is ready to take connections which causes the adapter to throw an exception. 
If you wait-on for rabbitmq in the datasource-integration test like it is done in the stateless test, the problem should be solved.

File: adapter/integration-test/src/adapter-datasource.test.js (hunk 2 above)
Comment (jenswaechtler) [reply adapter/integration-test/src/adapter-datasource.test.js]: The reasoning is that all data sources are treated as dynamic anyway. Either they have open parameters or not. But if you think that makes more sense, then it is fine with me.

**Review by mathiaszinnen — APPROVED** [review]

**Pull request merged**